MyLibrary/Sources/LiveTranslationFeature/LiveTranslation.swift

```diff
@@ -0,0 +1,594 @@
+
+@Reducer
+public struct LiveTranslation {
+  @ObservableState
+  public struct State: Equatable {
+    /// Live Translation Room Number
+    var roomNumber: String
+    /// Current visible translation items
+    var chatList: [TranslationEntity.CompositeChatItem] = []
+    /// Current language set
+    var langSet: LanguageEntity.Response.LangSet? = .none
+    /// Available language list
+    var langList: [LanguageEntity.Response.LanguageItem] = []
+    /// Live Translation Room Info
+    var roomInfo: ChatRoomEntity.Make.Response? = .none
+    /// Current language code which user selected
+    var selectedLangCode: String =
+      Locale.autoupdatingCurrent.language.languageCode?.identifier ?? "en"
+
+    /// While updating chat
+    var isUpdatingChat: Bool = false
+    /// While updating translation response
+    var isUpdatingTR: Bool = false
+    /// Chat updating request queue
+    var updateChatWaitingQueue: [RealTimeEntity.Chat.Response] = []
+    /// Translation response request queue
+    var updateTrWaitingQueue: [RealTimeEntity.Translation.Response] = []
+    /// Latest item's list type
+    var latestListType: RealTimeEntity.ListType? = .none
+
+    /// Streaming is connected
+    var isConnected: Bool = false
+    /// The task of connecting stream
+    var chatStreamTask: Task<Void, Never>? = nil
+
+    /// selected language sheet
+    var isSelectedLanguageSheet: Bool = false
+    /// showing last chat
+    var isShowingLastChat: Bool = false
+
+    public init(roomNumber: String) {
+      self.roomNumber = roomNumber
+    }
+  }
+
+  public enum Action: BindableAction, ViewAction {
+    case binding(BindingAction<State>)
+    case connectChatStream
+    case disconnectChatStream
+    case changeLangCode(String)
+    case view(View)
+
+    case handleResponseChat(RealTimeEntity.Chat.Response)
+    case checkUpdateChatWaitingQueue
+    case handleResponseTranslation(RealTimeEntity.Translation.Response)
+    case checkUpdateTRWaitingQueue
+
+    public enum View {
+      case onAppear
+      case connectStream
+      case disconnectStream
+      case selectLangCode(String)
+      case setSelectedLanguageSheet(Bool)
+      case setShowingLastChat(Bool)
+    }
+  }
+
+  @Dependency(\.liveTranslationServiceClient) var liveTranslationServiceClient
+
+  public init() {}
+
+  public var body: some ReducerOf<Self> {
+    BindingReducer()
+    Reduce { state, action in
+      switch action {
+      case .view(.onAppear):
+        return .run { [state] send in
+          await withTaskGroup(of: Void.self) { group in
+            group.addTask {
+              await loadLangSet(send: send)
+            }
+            group.addTask {
+              await loadChatRoomInfo(state: state, send: send)
+            }
+            group.addTask {
+              await loadLangList(send: send)
+            }
+          }
+        }
+      case .view(.connectStream):
+        return .run { send in
+          await send(.connectChatStream)
```

Comment:
Call function instead. Sending other action in send causes performance issue.

Reply:
@d-date 
Thank you. I rewrite it on https://github.com/tryswift/try-swift-tokyo/pull/97/commits/b32d9b470d83a5c43404f8100e3470bef2a7da77 .
By the way, is it okay to dispatch another action within a function that is itself called by an action, such as in the case of `handleResponseChat`?